### PR TITLE
feat(api): user memberships

### PR DESCRIPTION
**Pull request by TorstenDittmann**
## What does this PR do?

- adds an endpoint to list all memberships of a user
- adds new route to UI under a user
- changes response model for memberships:

```diff
userId
- name
- email
+ userName
+ userEmail
teamId
+ teamName
```
Bildschirmfoto 2022-05-12 um 15 23 13

## Test Plan

- adapted and added tests

## Related PRs and Issues

- #3046 

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

✅

### CODE REVIEW

File: app/views/console/users/team.phtml [hunk 1]
```diff
@@ -111,14 +111,14 @@
                                                 <button class="danger">Remove</button>
                                             </form>
 
-                                            <img src="" data-ls-attrs="src={{member|avatar}}" data-size="200" alt="User Avatar" class="avatar pull-start margin-end" loading="lazy" width="60" height="60" />
```
Comment (Meldiron) [inline app/views/console/users/team.phtml:114]: Not sure about this change. Why from `member` to `member.userName` Was `member` previously object, ID, email or name?

Comment (TorstenDittmann) [reply app/views/console/users/team.phtml:114]: https://github.com/appwrite/appwrite/blob/70efbf1fb9c3277043ef0255ba320b02beee3bbe/public/scripts/filters.js#L8

We have a fallback to the name property 👍🏻

File: app/controllers/api/teams.php [hunk 2]
```diff
@@ -423,8 +423,9 @@
 
         $response->setStatusCode(Response::STATUS_CODE_CREATED);
         $response->dynamic($membership
-            ->setAttribute('email', $email)
-            ->setAttribute('name', $name)
+            ->setAttribute('teamName', $team->getAttribute('name'))
+            ->setAttribute('userName', $user->getAttribute('name'))
+            ->setAttribute('userEmail', $user->getAttribute('email'))
```
Comment (Meldiron) [inline app/controllers/api/teams.php:428]: Previous:
```
->setAttribute('email', $email)
->setAttribute('name', $name)
```
            
Again just making sure, was this on purpose - not using variable?

File: docs/references/users/get-user-memberships.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Get the user membership list by its unique ID.
```
Comment (Meldiron) [inline docs/references/users/get-user-memberships.md:1]: Can we mention the limit? I can't see pagination, so there is some.

File: src/Appwrite/Utopia/Response/Model/Membership.php [hunk 4]
```diff
@@ -22,24 +22,30 @@ public function __construct()
...
+                'example' => 'john@appwrite.io',
+            ])
             ->addRule('teamId', [
                 'type' => self::TYPE_STRING,
                 'description' => 'Team ID.',
                 'default' => '',
                 'example' => '5e5ea5c16897e',
             ])
-            ->addRule('name', [
+            ->addRule('teamName', [
```
Comment (Meldiron) [inline src/Appwrite/Utopia/Response/Model/Membership.php:43]: Any reason why not to add teamID too? Could not be useful for us in UI, but might be really useful in server-SDK. Overall, if we mention some resources (like a team) and provide some specific data (name), we should also provide ID so they can get all other data of the team if they need to.

**Review by Meldiron — COMMENTED** [review]
Last comments, the rest looks good.

File: app/controllers/api/teams.php (hunk 2 above)
Comment (TorstenDittmann) [reply app/controllers/api/teams.php:428]: Yeah it was, there is so much happening before I would use `$name` again, that I wanted to be a 100% sure and use the object from the database 👍🏻

**Review by Meldiron — APPROVED** [review]

File: app/controllers/api/teams.php (hunk 2 above)
Comment (Cowtail1) [reply app/controllers/api/teams.php:428]: Response

**Pull request merged**